File: Dev/C++/Mascaret/src/VEHA/Behavior/StateMachine/PseudoState.cpp
#include "VEHA/Behavior/StateMachine/PseudoState.h"
#include "VEHA/Kernel/InstanceSpecification.h"

#include "VEHA/Behavior/Common/BehaviorExecution.h"
#include "VEHA/Behavior/Common/behaviorScheduler.h"


namespace VEHA
{

PseudoState::PseudoState(const string& name)
:Vertex(name)
{
}

PseudoState::~PseudoState()
{
}

shared_ptr<BehaviorExecution> PseudoState::activate(shared_ptr<InstanceSpecification> /*host*/, const Parameters & /*p*/)
{
	return shared_ptr<BehaviorExecution>();
}

void PseudoState::desactivate(void)
{
}

}

File: Dev/CS/Mascaret/Mascaret/VEHA/Behavior/StateMachine/PseudoState.cs

using System;
using System.Collections.Generic;

namespace Mascaret
{
    public enum PseudoStateKind
    { INITIAL = 0, DEEPHISTORY = 1, SHALLOWHISTORY = 2, JOIN = 3, FORK = 4, JUNCTION = 5, CHOICE = 6, ENTRYPOINT = 7, EXITPOINT = 8, TERMINATE = 9 };

    [Serializable]
    public class PseudoState : Vertex
    {

        public PseudoStateKind kind;


        public PseudoState(string name)
            : base(name)
        {
        }

        public override BehaviorExecution activate(InstanceSpecification host, Dictionary<string, ValueSpecification> p)
        {
            return null;
        }

        public override void desactivate()
        {

        }
    }
}

